test/test_render_html.py

```diff
@@ -142,3 +142,27 @@ def test_order_of_items() -> None:
     data = ["h1", "some ", ["span.pys", "<py>"]]
 
     assert render(data) == '<h1>some <span class="pys">&lt;py&gt;</span></h1>'
+
+
+def test_html_entities() -> None:
+    """HTML entities get escaped."""
+    data = ["p", "Go &larr; left or &rarr; right."]
+
+    assert render(data) == "<p>Go &amp;larr; left or &amp;rarr; right.</p>"
+
+
+def test_dangerously_set_inner_html() -> None:
+    """Mimick dangerouslySetInnerHTML behavior."""
+    data = [
+        "div",
+        [
+            "a",
+            {
+                "href": "/location",
+                "dangerouslySetInnerHTML": {"__html": "Location &rarr;"},
+            },
+            "Location",
+        ],
```

Comment:
I have a question: why not adding the `location &arr;` as a child html node to `a`?

I didn't know about the _dangerouslySetInnerHTML` before. Is that mainly for the React parser? I'm thinking if it would make sense to add the child directly. Please forgive me if I misunderstand!

Reply:
I think I understand the problem better, sorry for the noise! I will continue reviewing 😄